src/app.js

```diff
@@ -1 +1,41 @@
+/* eslint-disable no-console */
 'use strict';
+
+const { compareNumbers } = require('./compareNumber');
+const { generateNumber } = require('./generator');
+const { terminal } = require('./terminal');
+
+const generatedNumber = generateNumber();
+
+const checkUserNumber = (userNumber) => {
+  const isUserNumberValid = new Set(userNumber).size === 4
+  && !isNaN(userNumber);
+
+  if (!isUserNumberValid) {
+    console.log('Insert a number of 4 different digits');
+
+    playBullsAndCows();
+
+    return;
+  }
+
+  const { bull, cow } = compareNumbers(userNumber, generatedNumber);
+
+  if (bull === 4) {
+    console.log('U won, congratulations!');
+
+    terminal.close();
```

Comment:
You are missing return statement here, because after showing a user the success message and closing terminal you don't need to execute your code further. 
In your example, your game looks like a bully boy: at first it shows `U won` but then shows `Try again` and just when you are ready to try again it just breaks, without ever letting you to try :)
![image](https://user-images.githubusercontent.com/45945732/234862604-c010bf21-c7ce-44ea-b994-ff3dc54bde2c.png)